contracts/asset-proxy/contracts/src/bridges/MixinGasToken.sol

```diff
@@ -0,0 +1,41 @@
+/*
+
+  Copyright 2019 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.5.16;
+
+import "@0x/contracts-utils/contracts/src/DeploymentConstants.sol";
+import "../interfaces/IGasToken.sol";
+
+
+contract MixinGasToken is
+    DeploymentConstants
+ {
+
+    /// @dev Burns gas tokens based on the amount of gas consumed in the function
+    modifier burnsGasTokens {
+        uint256 gasBefore = gasleft();
+        _;
+        IGasToken gst = IGasToken(_getGstAddress());
+        if (address(gst) != address(0) && msg.sender == _getERC20BridgeProxyAddress()) {
```

Comment:
Do we actually need any of these checks? `address(gst) == address(0)` will probably only be true on testnets, and `msg.sender == _getERC20BridgeProxyAddress()` is usually enforced through different checks. This modifier may end up being useful elsewhere without these.

Reply:
Ah nvm, I guess we don't check if `msg.sender == _getERC20BridgeProxyAddress()` on all of the bridges.

Reply:
Do we want to these bridges (Uni,Oasis,Curve,Kyber) `bridgeTransferFrom` only callable via BridgeProxy address?

Happy to create a new modifier if we can guarantee that assumption

Reply:
I personally like these being permissionless.

Reply:
I've removed these checks, if a bridge wants em it can add em.